FreeRTOS/FreeRTOS
# Add MQTT keep-alive demo

MQTT keep-alive demo with QoS0

Description
-----------

The MQTT keep-alive demo is similar to the plaintext demo with QoS0. The only difference is that it creates an auto-reload timer that invokes a callback responsible for sending a PINGREQ packet. The timer is reset whenever a control packet is sent. Because the keep-alive mechanism is handled by the application, MQTT_ReceiveLoop is used in place of MQTT_ProcessLoop. 

Test Steps
-----------

Just open mqtt_keep_alive_demo.sln and build and run as you normally would.

Related Issue
-----------

N/A

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
